src/scenes/PlantsDashboardRouter/PlantsDashboardView.tsx

```diff
@@ -176,30 +116,31 @@ export default function PlantsDashboardView({ projectId, organizationId }: Plant
   }, [latestObservation]);
 
   const renderMortalityRate = useCallback(
-    () => (
-      <>
-        <Grid item xs={12}>
-          <Box
-            sx={{
-              display: 'flex',
-              alignItems: isMobile ? 'flex-start' : 'center',
-              flexDirection: isMobile ? 'column' : 'row',
-            }}
-          >
-            <Typography fontWeight={600} fontSize={'20px'} paddingRight={1}>
-              {strings.MORTALITY_RATE}
-            </Typography>
-            {hasObservations && (
-              <Typography>{strings.formatString(strings.AS_OF_X, getLatestObservationLink())}</Typography>
-            )}
-          </Box>
-        </Grid>
-        <Grid item xs={12}>
-          <MortalityRateCard plantingSiteId={selectedPlantingSiteId} />
-        </Grid>
-      </>
-    ),
-    [selectedPlantingSiteId, getLatestObservationLink, hasObservations]
+    () =>
+      plantingSite ? (
+        <>
+          <Grid item xs={12}>
+            <Box
+              sx={{
+                display: 'flex',
+                alignItems: isMobile ? 'flex-start' : 'center',
+                flexDirection: isMobile ? 'column' : 'row',
+              }}
+            >
+              <Typography fontWeight={600} fontSize={'20px'} paddingRight={1}>
+                {strings.MORTALITY_RATE}
+              </Typography>
+              {hasObservations && (
+                <Typography>{strings.formatString(strings.AS_OF_X, getLatestObservationLink())}</Typography>
+              )}
+            </Box>
+          </Grid>
+          <Grid item xs={12}>
+            <MortalityRateCard />
+          </Grid>
+        </>
+      ) : undefined,
+    [plantingSite, getLatestObservationLink, hasObservations]
   );
 
   const renderTotalPlantsAndSpecies = useCallback(
```

Comment:
If this doesn't have any dependencies anymore, does it still need the `useCallback`?